include/glow/Base/Tensor.h

```diff
@@ -559,18 +562,36 @@ class Tensor final {
   }
 
   template <class ElemTy>
-  bool isEqualImpl(const Tensor &other, float allowedError) const {
+  bool isEqualImpl(const Tensor &other, float allowedError,
+                   bool verbose) const {
     auto const *myData = getRawDataPointer<ElemTy>();
     auto const *otherData = other.getRawDataPointer<ElemTy>();
+    double maxErr = 0.0;
+    size_t maxErrIdx = 0, numExceedingError = 0;
     for (size_t i = 0, e = size(); i < e; i++) {
       double delta = myData[i] - otherData[i];
+      delta = std::abs(delta);
       // Since any comparison with NAN returns false, we use a negated condition
       // so that this function correctly returns false when delta is NAN.
-      if (!(std::abs(delta) <= allowedError)) {
-        return false;
+      if (!(delta <= allowedError)) {
+        if (!verbose) {
+          return false;
+        }
+        numExceedingError += 1;
+        if (maxErr < delta) {
+          maxErr = delta;
+          maxErrIdx = i;
+        }
       }
     }
-    return true;
+    if (numExceedingError != 0) {
```

Comment:
What happens if the only error is a NaN at index > 0? Seems like we'll increment numExceedingError for that, but the maxErr < delta check will fail and hence the maxErrIdx and maxErr won't be updated, right?

Reply:
Good call -- I should have taken a hint from the above NAN comment. Fixed!